### PR TITLE
[ᚬframework] feat(core/binding): add default StoreBool, StoreUint64, StoreString

### PR DESCRIPTION
**Pull request by mkxbl**
**What type of PR is this?**
Add feat(core/binding)
**What this PR does / why we need it**:
Add DefaultStoreBool, DefaultStoreUint64, DefaultStoreString
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: core/binding/src/store/tests/primitive.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
+use std::sync::Arc;
```
Comment (yejiayu) [inline core/binding/src/store/tests/primitive.rs]: move to core/binding/src/tests

File: core/binding/src/store/primitive.rs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        }
+    }
+}
+
+impl<DB: TrieDB> StoreBool for DefaultStoreBool<DB> {
+    fn get(&self) -> ProtocolResult<bool> {
+        let opt_bs: Option<Bytes> = self.state.get(&self.key)?;
+
+        match opt_bs {
+            Some(bs) => {
```
Comment (yejiayu) [inline core/binding/src/store/primitive.rs]: let bs = opt_bs.ok_or(BindingError::Store(StoreType::GetNone).into())?;

File: core/binding/src/store/primitive.rs [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+                let mut rdr = Cursor::new(bs.to_vec());
+                let u = rdr.read_u8().unwrap();
```
Comment (yejiayu) [inline core/binding/src/store/primitive.rs]: Don't use `unwrap`, if it never happens you can use `expect` with some description.

File: core/binding/src/store/primitive.rs [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+}
+
+impl<DB: TrieDB> StoreUint64 for DefaultStoreUint64<DB> {
+    fn get(&self) -> ProtocolResult<u64> {
+        let opt_bs: Option<Bytes> = self.state.get(&self.key)?;
+
+        match opt_bs {
+            Some(bs) => {
+                let mut rdr = Cursor::new(bs.to_vec());
+                Ok(rdr.read_u64::<BigEndian>().unwrap())
```
Comment (yejiayu) [inline core/binding/src/store/primitive.rs]: Why use Big-endian ?

File: core/binding/src/store/primitive.rs [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+        self.state.insert(self.key.clone(), val)?;
+        Ok(())
+    }
+
+    fn get(&self) -> ProtocolResult<String> {
+        let opt_bs: Option<Bytes> = self.state.get(&self.key)?;
+
+        match opt_bs {
+            Some(bs) => Ok(String::from_utf8(bs.to_vec()).unwrap()),
```
Comment (yejiayu) [inline core/binding/src/store/primitive.rs]: try use ok_or

File: core/binding/src/store/tests/primitive.rs (hunk 1 above)
Comment (mkxbl) [reply core/binding/src/store/tests/primitive.rs]: done

File: core/binding/src/store/primitive.rs [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+use crate::store::StoreType;
+use crate::BindingError;
+
+pub struct DefaultStoreBool<DB: TrieDB> {
+    state: GeneralServiceState<DB>,
+    key:   Hash,
+}
+
+impl<DB: TrieDB> DefaultStoreBool<DB> {
+    pub fn new(state: GeneralServiceState<DB>, var_name: &str) -> Self {
```
Comment (yejiayu) [inline core/binding/src/store/primitive.rs]: Should not pass a concrete `ServiceState` implementation.

In addition, `GeneralServiceState` does not currently support `clone`. How does it share state between modules?

File: core/binding/src/store/mod.rs [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+mod map;
+mod primitive;
+
+use derive_more::{Display, From};
+
+pub use primitive::{DefaultStoreBool, DefaultStoreString, DefaultStoreUint64};
+
+#[derive(Debug, Display, From)]
+pub enum StoreType {
+    // The key not existed
```
Comment (yejiayu) [inline core/binding/src/store/mod.rs]: Consider implementing a StoreError

**Comment by mkxbl** [comment]
ping@yejiayu

**Comment by mkxbl** [comment]
ping@yejiayu

File: core/binding/src/store/primitive.rs (hunk 6 above)
Comment (mkxbl) [reply core/binding/src/store/primitive.rs]: Changed to `pub struct DefaultStoreBool<S: ServiceState>` and use `state : Rc<RefCell<S>>`

**Pull request merged**